tasks.py

```diff
@@ -689,10 +724,67 @@ def ci_dnsdist_configure(c, features):
         '--prefix=/opt/dnsdist'
     ])
 
-    res = c.run(configure_cmd, warn=True)
-    if res.exited != 0:
-        c.run('cat config.log')
-        raise UnexpectedExit(res)
+def ci_dnsdist_configure_meson(features, additional_flags, build_dir):
+    if features == 'full':
+      features_set = '-D cdb=enabled \
+                      -D dnscrypt=enabled \
+                      -D dnstap=enabled \
+                      -D ebpf=enabled \
+                      -D h2o=enabled \
+                      -D ipcipher=enabled \
+                      -D libedit=enabled \
+                      -D libsodium=enabled \
+                      -D lmdb=enabled \
+                      -D nghttp2=enabled \
+                      -D re2=enabled \
+                      -D systemd=enabled \
+                      -D tls-gnutls=enabled \
+                      -D dns-over-https=true \
+                      -D dns-over-http3=true \
+                      -D dns-over-quic=true \
+                      -D dns-over-tls=true \
+                      -D reproducible=true \
+                      -D snmp=true'
+    else:
+      features_set = '-D cdb=disabled \
+                      -D dnscrypt=disabled \
+                      -D dnstap=disabled \
+                      -D ebpf=disabled \
+                      -D h2o=disabled \
+                      -D ipcipher=disabled \
+                      -D libedit=disabled \
+                      -D libsodium=disabled \
+                      -D lmdb=disabled \
+                      -D nghttp2=disabled \
+                      -D re2=disabled \
+                      -D systemd=disabled \
+                      -D tls-gnutls=disabled \
+                      -D dns-over-https=false \
+                      -D dns-over-http3=false \
+                      -D dns-over-quic=false \
+                      -D dns-over-tls=false \
+                      -D reproducible=false \
+                      -D snmp=false'
+    unittests = get_unit_tests(meson=True)
+    fuzztargets = get_fuzzing_targets(meson=True)
+    tools = f'''AR=llvm-ar-{clang_version} RANLIB=llvm-ranlib-{clang_version}''' if is_compiler_clang() else ''
+    cflags = " ".join([get_cflags()])
+    cxxflags = " ".join([get_cxxflags(), additional_flags])
+    return " ".join([
+        tools,
+        f'CFLAGS="{cflags}"',
+        f'CXXFLAGS="{cxxflags}"',
+        f"CC='{get_c_compiler()}'",
+        f"CXX='{get_cxx_compiler()}'",
+        f'. {repo_home}/.venv/bin/activate && meson setup {build_dir}',
```

Comment:
```suggestion
        f'{repo_home}/.venv/bin/meson setup {build_dir}',
```

otherwise you don't pick up any of the env vars from the start of the line

Reply:
This works better:
```
@@ -469,12 +452,14 @@ def get_base_configure_cmd(additional_c_flags='', additional_cxx_flags='', enabl
 def get_base_configure_cmd_meson(build_dir, additional_c_flags='', additional_cxx_flags='', enable_systemd=True, enable_sodium=True):
     cflags = " ".join([get_cflags(), additional_c_flags])
     cxxflags = " ".join([get_cxxflags(), additional_cxx_flags])
-    return " ".join([
+    env = " ".join([
         f'CFLAGS="{cflags}"',
         f'CXXFLAGS="{cxxflags}"',
         f"CC='{get_c_compiler()}'",
-        f"CXX='{get_cxx_compiler()}'",
-        f'. {repo_home}/.venv/bin/activate && meson setup {build_dir}',
+        f"CXX='{get_cxx_compiler()}'"
+    ])
+    return " ".join([
+        f'. {repo_home}/.venv/bin/activate && {env} meson setup {build_dir}',
         "-D systemd={}".format("enabled" if enable_systemd else "disabled"),
         "-D signers-libsodium={}".format("enabled" if enable_sodium else "disabled"),
         "-D hardening-fortify-source=auto",
```